cesium-ml/cesium_web
# Sticky notification for websocket connection problems

## Overview
Creates a sticky notification when the websocket is not available, or when it is not authenticated (slightly different message). The notification is tied to the websocket status and not to the tricolor status (so there is no flicker on load). The websocket notification is cleared before a new one is added, so that there is always at most one websocket notification. 

## Testing: 
### Testing connection issues:
- Run `make`
- Check that no notification is up
- Run `supervisorctl -c conf/supervisord_common.conf`
  - Run `stop websocket`
  - Check that there is a sticky notification
  - Run `start websocket`
  - Check that the notification is gone

### Testing authentication issues:
- Run `make`
- Run `supervisorctl -c conf/supervisord_common.conf`
- Edit `websocket_server.py` function `authenticate` to always return `AUTH_FAILED`
- restart the websocket server with `restart websocket`
- check that there is a sticky notification and the status is orange
- undo the changes and restart the websocket server
- check that the notification leaves (once the websocket server reloaded)
